src/features/devices/models/device-additions.ts

```diff
@@ -0,0 +1,367 @@
+import {
+	AbstractSqlModel,
+	AbstractSqlQuery,
+	EqualsNode,
+} from '@balena/abstract-sql-compiler';
+
+export const addToModel = (
+	abstractSql: AbstractSqlModel,
+	shimActiveField: boolean = true,
+) => {
+	// FIXME: this is a shim for the `is web accessible` attribute on the device
+	// resource. The core API has no support for "Device Public URLs".
+	abstractSql.tables['device'].fields.push({
+		fieldName: 'is web accessible',
+		dataType: 'Boolean',
+		// The cast is needed because AbstractSqlQuery cannot express a constant value.
+		computed: ['Boolean', false] as AbstractSqlQuery,
+	});
+
+	// FIXME: cloud has the notion of active/inactive devices via a field
+	// on the device resource. Computing the overall status below depends
+	// on whether the device is active but also on a few other statuses
+	// (Ordered, Preparing, etc.) that make no sense for the core API.
+	//
+	// We must somehow expose the overall status/progress attributes on the
+	// model and we could contrive a way for the cloud to inject the extra
+	// cases, but that would quickly become overly complex because the order
+	// of cases matters. We'd never be able to cleanly make the core API
+	// agnostic to the cloud.
+	//
+	// Therefore, I opted for the most straightforward way, which is to switch
+	// the computation based on a flag. When `shimActiveField` is true (the
+	// default), it'll generate a dummy case (the equivalent of 1 == 2, which is
+	// always false), otherwise it'll generate a case that looks into the actual
+	// model field.
+	const isInactive: EqualsNode = shimActiveField
+		? ['Equals', ['Number', 1], ['Number', 2]]
```

Comment:
Can this be shimmed in the same way as `is web accessible`?

Reply:
@Page- do you mean the specific case where `shimActiveField` is `true`? It would be nice, but the types are incompatible -- `isInactive` is an `EqualsNode` and `is web accessible` is cast to `AbstractSqlQuery`.

Reply:
I guess I could type `isInactive` as `AbstractSqlQuery`?

Reply:
I meant something like
```ts
	abstractSql.tables['device'].fields.push({
		fieldName: 'is active',
		dataType: 'Boolean',
		computed: ['Boolean', true],
	});
```
and then you don't need the shim here at all

Reply:
@Page- I tried that before but doesn't work -- the field isn't being replaced in the query, so it fails with:

```
DatabaseError: column device.is active does not exist
```

Which is very unfortunate because this is the most elegant way to provide model-level default behaviour in core API.